docs/misc/message_system.md

```diff
@@ -0,0 +1,242 @@
+# Message System
+
+Message system was implemented to allow sending emergency messages to Trezor Suite app to a user with specific stack.
+
+[Example messages](https://www.notion.so/1d622e7079c14f64a03fa17f50bce14f?v=b8649de184c3437abb983990c5513e5f)
+
+[Issue on Github](https://github.com/trezor/trezor-suite/issues/2752)
+
+## Types of in-app messages
+
+There are four ways of displaying message to a user.
+
+- banner
+   - looks like a cookie bar above the page
+- modal
+   - *TODO: missing implementation*
+- context 
+   - messages on specific places in app - high level (e.g. settings page)
+   - *TODO: missing implementation*
+- super-context 
+   - messages on specific places in the app - low level (e.g. category in settings page)
+   - *TODO: missing implementation*
+
+## Implementation
+
+### Config
+
+The system of messages is based on a configuration file in which messages with specific conditions ​are described. If specific conditions are satisfied, the message is shown to a user.
+
+Current configuration file is located in `packages/suite-data/src/message-system/config`. Its name is `config.vX.json`. The `X` express current version messaging system.
+
+The config is fetched at launch of the application and then every six hours (If the application is running). It remembers the previously fetched config to inform the user even if he is offline. For this reason, the latest available config during build time is bundled with the application.
+
+If fetching of a new config fails, the fetching process is repeated every 10 minutes.
```

Comment:
I am wondering if this is not too aggressive. Imagine we fuck up, remove the config from our server and then thousands of Suites are talking to our server each 10 minutes. It shouldn't be a problem but to be on the safe side maybe one hour is enough?

Reply:
changed